release/models/platform/openconfig-platform.yang

```diff
@@ -540,11 +546,13 @@ module openconfig-platform {
     }
 
     leaf model-name {
+      when "../removable = 'true'";
       type string;
       description
-        "A human readable string describing the model of a component.
-        This string is optional and should only be populated if part-no
-        is also populated.";
+        "This is the component name that would be found in a catalog of 
+        stock keeping units (SKU) and should be the orderable name of
+        the component. This is required for all FRU (field replaceable unit)
```

Comment:
Note: due to the conditional when statement, this leaf will _only_ be available for components with `removable=true` and never set for any other component.  I think you are just stating here that this leaf is mandatory in the description for such cases vs. setting `mandatory true;` to enforce.  Any reason not to set the `mandatory` stmt here? (This will then _only_ be mandatory when removable=true and not mandatory otherwise)

Nits: 
* `"This is the component name..."` sounds a bit confusing since this terminology is used for the actual key name itself.  Any reason you didn't just use `model name` here?
* For the path `s/TRUE/true/`
* Another confusing point is the there is a catchall identity today for the `type` named `FRU`.  I recall we spoke about this in a PR long back as to it's potential removal but it is likely there are various implementations where the `type` identities is not fully comprehensive thus folks may set to `FRU` with the `removable` leaf set to `false` (not really removable but no other identity suits without creating vast augments).  Something we should probably clean up in a separate PR to shape this and reduce further confusion

Reply:
Thanks for the feedback! I have set mandatory true so that it is mandatory whenever removable is true and is optional otherwise (at least I believe that is what my implementation is now doing).

@earies what do you mean by your comment "For the path s/True/true/"?

I decided to remove the comment about it being required for FRUs from the description because I believe making it mandatory when removable is true achieves the same thing, and as you mentioned might be more accurate since the type FRU is sometimes "misused". @dplore does this sound ok to you to use removable=true instead of type=FRU?

Reply:
> @earies what do you mean by your comment "For the path s/True/true/"?

The lexical representation for a boolean is represented as lower case true/false is all I mean here

https://datatracker.ietf.org/doc/html/rfc6020#section-9.5.1